eng/pipelines/mirror-repos.yml

```diff
@@ -11,61 +11,62 @@ jobs:
     displayName: Sync public repos
 
     steps:
+    - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
     - checkout: none
     - template: ./templates/steps/sync-repo-branch.yml
       parameters:
         GH_TOKEN: $(azuresdk-github-pat)
         Repos:
 
           Azure/azure-rest-api-specs:
-            Branch: master
+            Branch: $(DefaultBranch)
```

Comment:
For this mirror pipeline I think your change is probably Ok for now but once we add in the main/master we might want to leverage it to keep both master and main in-sync for a while. It all depends on how we plan to transition things.

Reply:
Actually come to think about it we might need to change this logic because each of these repo's might have different default branches. We likely need to implement this in the sync-repo-branch template to pull the default branch for the specific repo.

Reply:
We plan to switch default branch master to main through the Github UI. We can have many supporting features by doing this. Like the master link will not fail. PR will target to main instead of master.

So only one branch can exist (master or main) based on the transition plan.

Reply:
Correct @sima-zhu, originally the GitHub rename feature didn't exist. Now that it does I think it simplifies things a bit. But cross repo coordination is still a challenge (e.g. we should probably change both the public and -pr repos for each pair simultaneously.

Reply:
Since the `workingDirectory` is for `azure-sdk-tools`, move to the place we actually have the lang repo inside of sync-repo-branch.yml